lib/mm_network.pm

```diff
@@ -70,7 +71,8 @@ sub configure_dhcp {
     enter_cmd 'done';
     save_screenshot;
     assert_script_run "rcnetwork restart";
-    assert_script_run "ip addr";
+    assert_script_run "ip link show \$NIC | grep -A9 'UP'";
+    script_retry "ip addr show \$NIC | grep -B9 -A9 'inet '", delay => 15, retry => 12;
```

Comment:
Nice implementation, you even took care of not getting a false assertion in case of only having inet6